js/preload/preloadMain.js

```diff
@@ -121,6 +123,28 @@ function createAPI() {
         },
 
         /**
+         * Open file in default app.
+         */
+        openFile: function (id) {
+            let fileIndex = _.findIndex(local.downloadItems, {_id: id});
```

Comment:
I had skipped this because download items are being controlled by us. Either way, it makes sense to have a safety check. I've added it now.